scripts/asm-installer/vsphere/attach_vm

```diff
@@ -0,0 +1,219 @@
+  CONTEXT=${currContext:10}
+}
+
+read_args() {
+  while [[ ${#} != 0 ]]; do
+    case "${1}" in
+    -addr | --address | --vm_ip | --vm-ip)
+      export VM_IP="${2}"
+      shift 2
+      ;;
+    -app | --app | --vm_app | --vm-app)
+      export VM_APP="${2}"
+      shift 2
+      ;;
+    -mesh | --mesh_id | --mesh-id)
+      export MESH_ID="${2}"
+      shift 2
+      ;;
+    -network | --network | --vm_network | --vm-network)
+      export VM_NETWORK="${2}"
+      shift 2
+      ;;
+    -k | --key_file | --ssh-key-file)
+      export KEY_FILE="${2}"
+      shift 2
+      ;;
+    -s | --service_account | --service-account)
+      export SERVICE_ACCOUNT="${2}"
+      shift 2
+      ;;
+    -l | --cluster_location | --cluster-location)
+      export CLUSTER_LOCATION="${2}"
+      shift 2
+      ;;
+    -n | --cluster | --cluster_name | --cluster-name)
+      export CLUSTER="${2}"
+      shift 2
+      ;;
+    -p | --project_id | --project-id)
+      export PROJECT_ID="${2}"
+      shift 2
+      ;;
+    -ns | --namespace)
+      export VM_NAMESPACE="${2}"
+      shift 2
+      ;;
+    *)
+      warn "Unknown option ${1}"
+      exit 2
+      ;;
+    esac
+  done
+}
+
+prepare_vm() {
+  cd ${WORK_DIR}
+
+  scp -i "${KEY_FILE}" -o StrictHostKeyChecking=no runagent.sh root@${VM_IP}:~
+#  scp -i "${KEY_FILE}" -o StrictHostKeyChecking=no istio-sidecar.rpm root@${VM_IP}:~
+#  scp -i "${KEY_FILE}" -o StrictHostKeyChecking=no server root@${VM_IP}:~
+
+  if [[ -e ${OUTPUT_DIR}/vm-${VM_APP} ]]; then
+    rm -rf ${OUTPUT_DIR}/vm-${VM_APP}
+  fi
+
+  mkdir -p ${OUTPUT_DIR}/vm-${VM_APP}
+  cd ${OUTPUT_DIR}/vm-${VM_APP}
+
+}
+
+create_workloadgroup() {
+  # Register VM and create a service for the VM
+cat <<EOF > workloadgroup.yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: WorkloadGroup
+metadata:
+  name: "${VM_APP}"
+  namespace: "${VM_NAMESPACE}"
+spec:
+  metadata:
+    labels:
+      app: "${VM_APP}"
+  template:
+    serviceAccount: "${SERVICE_ACCOUNT}"
+    network: "${VM_NETWORK}"
+EOF
+
+  kubectl --namespace "${VM_NAMESPACE}" apply -f workloadgroup.yaml
+}
+
+create_workloadentry() {
+  cat <<EOF >k8sservice.yaml
+  apiVersion: v1
+  kind: Service
+  metadata:
+    name: "${VM_APP}"
+    namespace: "${VM_NAMESPACE}"
+    labels:
+      asm_resource_type: VM
+  spec:
+    ports:
+    - port: 8080
+      name: http
+      targetPort: 8080
+    selector:
+      service.istio.io/canonical-name: "${VM_APP}"
+EOF
+
+  kubectl --namespace "${VM_NAMESPACE}" apply -f k8sservice.yaml
+}
+
+register_vm_transfer_file() {
+  # Create files to transfer to VM
+  cd ${WORK_DIR}
+
+#--kubeconfig=$HOME/.kube/config
+  ./istioctl x workload entry configure -f ${OUTPUT_DIR}/vm-${VM_APP}/workloadgroup.yaml -o ${OUTPUT_DIR}/vm-${VM_APP} --clusterID ${CLUSTER} --autoregister
+
+  cd ${OUTPUT_DIR}/vm-${VM_APP}
+
+  scp -i "${KEY_FILE}" -o StrictHostKeyChecking=no cluster.env hosts istio-token mesh.yaml root-cert.pem root@${VM_IP}:~
```

Comment:
Let's not use home directory for storing the temporary files. Instead, create a /tmp directory if it does exist, and put the files there.